packages/backend/server/src/core/permission/types.ts

```diff
@@ -1,11 +1,287 @@
-export enum Permission {
-  Read = 0,
-  Write = 1,
-  Admin = 10,
-  Owner = 99,
-}
+import assert from 'node:assert';
 
 export enum PublicPageMode {
   Page,
   Edgeless,
 }
+
+export enum DocRole {
+  External = 0,
+  Reader = 10,
+  Editor = 20,
+  Manager = 30,
+  Owner = 99,
+}
+
+export enum WorkspaceRole {
+  External = -99,
+  Collaborator = 1,
+  Admin = 10,
+  Owner = 99,
+}
+
+export const Actions = {
+  Workspace: {
+    Sync: 1,
```

Comment:
What are the numbers meaning？

Reply:
It refers to the action index in `ActionsKeys`. Generally, a higher number indicates a more advanced role is required.

Reply:
this implementation is too dirty, we need to clean it up later